File: 30days/day8.ts
/*
 * Day 8: Dictionaries and Maps
 */
function main() {
	const n: number = parseInt(readLine());
	const phoneList: any = {};
	for (let i = 0; i < n; i++) {
		const stringNumber: string[] = readLine().split(' ');
		phoneList[stringNumber[0]] = stringNumber[1];
	}
	for (let i = 0; i < n; i++) {
		const stringName: string = readLine();
		if (stringName in phoneList) {
			console.log(`${stringName}=${phoneList[stringName]}`);
		} else {
			console.log('Not found');
		}
	}
}

File: HackerRank/30days/go/day8.go
/*
* Day 8: Dictionaries and Maps
 */
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()

	n, _ := strconv.Atoi(scanner.Text())
	data := make(map[string]string)

	for i := 0; i < n; i++ {
		scanner.Scan()
		arr_temp := strings.Split(scanner.Text(), " ")
		data[arr_temp[0]] = arr_temp[1]
	}

	for scanner.Scan() {
		if data[scanner.Text()] == "" {
			fmt.Println("Not found")
		} else {
			fmt.Printf("%s=%s", scanner.Text(), data[scanner.Text()])
		}
	}
}
